gallery/visualtests/nxTabs.js

```diff
@@ -66,5 +89,239 @@ describe('NxTabs', function() {
     it('looks right', simpleModalTest(tabModalNoHeaderExampleSelector));
   });
 
+  describe('keyboard navigation', function() {
+    it('focuses on the tab when it is clicked', async function() {
+      const [example] = await waitAndGetElements(tabManualActivationModeExampleSelector);
+
+      const secondTab = await example.$('.nx-tab:nth-child(2)');
+
+      expect(await isFocused(secondTab)).toBe(false);
+
+      await secondTab.click();
+
+      expect(await isFocused(secondTab)).toBe(true);
+    });
+
+    it('focuses on the nx-tab-panel and link inside the panel after tabbing from nx-tab', async function() {
+      const [example] = await waitAndGetElements(tabManualActivationModeExampleSelector);
+      const page = getPage();
+
+      const activeTab = await example.$('.nx-tab:nth-child(1)');
+      const tabPanel = await example.$('.nx-tab-panel');
+
+      await activeTab.focus();
+
+      expect(await isFocused(activeTab)).toBe(true);
+
+      await page.keyboard.press('Tab');
+
+      expect(await isFocused(tabPanel)).toBe(true);
+
+      const tabPanelAnchor = await example.$('.nx-tab-panel a');
+
+      await page.keyboard.press('Tab');
+
+      expect(await isFocused(tabPanelAnchor)).toBe(true);
+    });
+
+    it('focuses back on active nx-tab from nx-tab-panel when shift+tab is activated', async function() {
+      const [example] = await waitAndGetElements(tabManualActivationModeExampleSelector);
+      const page = getPage();
+
+      const activeTab = await example.$('.nx-tab:nth-child(1)');
+      const tabPanel = await example.$('.nx-tab-panel');
+
+      await activeTab.focus();
+
+      expect(await isFocused(activeTab)).toBe(true);
+
+      await page.keyboard.press('Tab');
+
+      expect(await isFocused(tabPanel)).toBe(true);
+
+      await page.keyboard.down('Shift');
+      await page.keyboard.press('Tab');
+      await page.keyboard.up('Shift');
+
+      expect(await isFocused(activeTab)).toBe(true);
+    });
+
+    it('focuses on the correct tab when arrow right is pressed but '
+      + 'does not activate it when in manual activation mode', async function() {
+      const [example] = await waitAndGetElements(tabManualActivationModeExampleSelector);
+      const page = getPage();
+
+      const firstTab = await example.$('.nx-tab:nth-child(1)');
+      const thirdTab = await example.$('.nx-tab:nth-child(3)');
+      const fourthTab = await example.$('.nx-tab:nth-child(4)');
+
+      await thirdTab.click();
+
+      expect(await isFocused(thirdTab)).toBe(true);
+      expect(await hasClass(thirdTab, 'active')).toBe(true);
+
+      await page.keyboard.press('ArrowRight');
+
+      expect(await isFocused(fourthTab)).toBe(true);
+      expect(await hasClass(fourthTab, 'active')).toBe(false);
+
+      await page.keyboard.press('ArrowRight');
+
+      expect(await isFocused(firstTab)).toBe(true);
+      expect(await hasClass(firstTab, 'active')).toBe(false);
```

Comment:
I'd say that this test and the next one should check which tab panel is visible/active during each step as well

Reply:
> I'd say that this test and the next few should check which tab panel is visible/active during each step as well

Updated the tests here: 08d960bd4908a06aea14a276fd7495f6069a2433